### PR TITLE
[G&M] React Query Properties

### PR DESCRIPTION
**Pull request by Guillar1**

### CODE REVIEW

File: packages/itwin/grouping-mapping-widget/src/components/context/PropertiesContext.ts [hunk 1]
```diff
@@ -3,24 +3,14 @@
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
 import * as React from "react";
-import type { CalculatedProperty, CustomCalculation, GroupProperty } from "@itwin/insights-client";
 
 export interface Properties {
```
Comment (MindaugasButkus) [inline packages/itwin/grouping-mapping-widget/src/components/context/PropertiesContext.ts:7]: Interface name no longer makes sense.

File: packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyTable.tsx [hunk 2]
```diff
@@ -42,83 +43,82 @@ export const GroupPropertyTable = ({
...
-          },
-        ],
+                  Remove
+                </MenuItem>,
+              ].flatMap((p) => p)}
+            >
+              <IconButton styleType='borderless' title='Property Options'>
+                <SvgMore
+                  style={{
+                    width: "16px",
```
Comment (MindaugasButkus) [inline packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyTable.tsx:96]: Is this styling still needed?

File: packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyAction.tsx [hunk 3]
```diff
@@ -73,121 +73,121 @@ export const GroupPropertyAction = ({
...
-          setPropertyName(response.propertyName);
-          setOldPropertyName(response.propertyName);
-          setDataType(response.dataType);
-          setQuantityType(response.quantityType);
-          const properties = findProperties(response.ecProperties, propertiesMetaData);
-          if (properties.length === 0) {
-            setPropertiesNotFoundAlert(true);
-          }
+    if (data.groupPropertyDetails) {
+      const { propertyName, dataType, quantityType, ecProperties } = data.groupPropertyDetails;
```
Comment (MindaugasButkus) [inline packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyAction.tsx:120]: I don't think it's worth deconstructing an object here. Are the individual properties of the object used in more than 1 place?

File: packages/itwin/grouping-mapping-widget/src/components/Properties/GroupColorToggle.tsx [hunk 4]
```diff
@@ -22,45 +25,52 @@ export const GroupColorToggle = ({
...
   const { iModelConnection } = useGroupingMappingApiConfig();
   if (!iModelConnection) {
     throw new Error("This component requires an active iModelConnection.");
   }
   const { showGroupColor, setShowGroupColor } = usePropertiesContext();
-  const { data: hiliteIdsResult } = useGroupKeySetQuery(group, iModelConnection, showGroupColor);
+  const { data: hiliteIdsResult, isFetched, isFetching } = useGroupKeySetQuery(group, iModelConnection, showGroupColor);
+  const isMounted = useIsMounted();
+
+  const { mutate: visualizationMutate, isLoading: isVisualizing } = useMutation({
```
Comment (MindaugasButkus) [inline packages/itwin/grouping-mapping-widget/src/components/Properties/GroupColorToggle.tsx:36]: `mutateVisualization`, not `visualizationMutate`.

File: packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyAction.tsx (hunk 3 above)
Comment (Guillar1) [reply packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyAction.tsx]: Nope. Changed. Also cleaned it up further. I was using a deprecated useQuery callback.

File: packages/itwin/grouping-mapping-widget/src/components/Properties/GroupColorToggle.tsx (hunk 4 above)
Comment (Guillar1) [reply packages/itwin/grouping-mapping-widget/src/components/Properties/GroupColorToggle.tsx]: Changed

**Review by MindaugasButkus — APPROVED** [review]

**Pull request merged**